src/main/java/com/project/service/UserServiceImpl.java

```diff
@@ -38,17 +39,23 @@ public User getById(final long id) throws UserNotFoundException {
         return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
     }
 
+    @Override
+    public User getByEmail(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException(email));
+    }
+
     @Override
     public User addUser(User user) throws UserException {
         userValidator.validate(user);
 
-        user.setId(null);
-
-        user.setPassword(bcryptEncoder.encode(user.getPassword()));
         if (userRepository.findByEmail(user.getEmail()).isPresent()) {
             throw new UserEmailExistsException(user.getEmail());
         }
 
+        user.setId(null);
```

Comment:
You should create a `Request` class then you won't need to do these manual id mutations. It's good to separate your DB model from your request/response objects.